### PR TITLE
Replace unsecure string operation in oecert

### PR DESCRIPTION
**Pull request by qiucwang**
`strcpy` and `strcat` are not secure. Replaced it with snprintf to hardcode enclave filename.
Signed-off-by: Qiucheng Wang <qiucwang@microsoft.com>

### CODE REVIEW

**Review by yentsanglee — APPROVED** [review]

File: tools/oecert/host/host.cpp [hunk 1]
```diff
@@ -250,18 +251,20 @@ static char* _get_enclave_filename()
...
 
     if (enclave_filename == nullptr)
     {
         printf("Failed to malloc enclave_filename.\n");
         goto done;
     }
-
-    strcpy(enclave_filename, path);
-    strcat(enclave_filename, ENCLAVE_FILENAME_SUFFIX);
+    snprintf(
```
Comment (dthaler) [inline tools/oecert/host/host.cpp:262]: This is still not safe, since `snprintf()` does not guarantee null termination.  
See https://en.cppreference.com/w/c/io/fprintf and https://stackoverflow.com/questions/1270387/are-snprintf-and-friends-safe-to-use
`snprintf_s()` on the other hand does safe truncation.

File: tools/oecert/host/evidence.cpp [hunk 2]
```diff
@@ -131,30 +131,30 @@ void parse_certificate_extension(const uint8_t* data, size_t data_len)
     printf(
         "\n    parsed qe certificate extension (%s) {\n",
         SGX_EXTENSION_OID_STR);
-    printf("        ppid (hex): ");
+    printf("      ppid (hex): ");
```
Comment (dthaler) [inline tools/oecert/host/evidence.cpp:134]: All the changes in this file look unrelated to the description of this PR.
https://github.com/openenclave/openenclave/blob/master/docs/Community/Contributing.md says:

> * **DON'T** mix independent, unrelated changes in one PR. Separate real project/test code changes from larger code formatting/dead code removal changes. Separate unrelated fixes into separate PRs, especially if they are in different libraries.

File: tools/oecert/host/host.cpp (hunk 1 above)
Comment (qiucwang) [reply tools/oecert/host/host.cpp:262]: 👍

File: tools/oecert/host/evidence.cpp (hunk 2 above)
Comment (qiucwang) [reply tools/oecert/host/evidence.cpp:134]: I'll put them in another PR.

File: tools/oecert/host/host.cpp (hunk 1 above)
Comment (qiucwang) [reply tools/oecert/host/host.cpp:262]: I checked again and `snprintf()` guarantees null termination. Microsoft's `_snprintf()` does not though.
https://stackoverflow.com/questions/7706936/is-snprintf-always-null-terminating
I also tested it on both linux and windows by truncating the buffer size, and the null terminator was included at both platforms.

**Comment by qiucwang** [comment]
bors try

**Review by dthaler — APPROVED** [review]
